packages/frontend/src/hooks/useOrganizationGroups.ts

```diff
@@ -1,62 +1,111 @@
 import {
     type ApiError,
+    type ApiGroupListResponse,
     type CreateGroup,
     type Group,
     type GroupWithMembers,
+    type KnexPaginateArgs,
     type UpdateGroupWithMembers,
 } from '@lightdash/common';
 import {
+    useInfiniteQuery,
     useMutation,
     useQuery,
     useQueryClient,
+    type UseInfiniteQueryOptions,
     type UseQueryOptions,
 } from '@tanstack/react-query';
-import Fuse from 'fuse.js';
 import { lightdashApi } from '../api';
 import useToaster from './toaster/useToaster';
 import useQueryError from './useQueryError';
 
-const getOrganizationGroupsQuery = async (includeMembers?: number) =>
-    lightdashApi<GroupWithMembers[]>({
-        url: `/org/groups${
-            includeMembers ? `?includeMembers=${includeMembers}` : ''
-        }`,
+const getOrganizationGroupsQuery = async (
+    includeMembers?: number,
+    searchQuery?: string,
+    paginateArgs?: KnexPaginateArgs,
+) => {
+    const urlParams = new URLSearchParams({
+        ...(paginateArgs
+            ? {
+                  page: String(paginateArgs.page),
+                  pageSize: String(paginateArgs.pageSize),
+              }
+            : {}),
+        ...(includeMembers ? { includeMembers: String(includeMembers) } : {}),
+        ...(searchQuery ? { searchQuery } : {}),
+    }).toString();
+
+    return lightdashApi<ApiGroupListResponse['results']>({
+        url: `/org/groups${urlParams ? `?${urlParams}` : ''}`,
         method: 'GET',
         body: undefined,
     });
+};
 
 export const useOrganizationGroups = ({
-    search,
+    searchInput,
     includeMembers,
     queryOptions,
 }: {
-    search?: string;
+    searchInput?: string;
     includeMembers?: number;
-    queryOptions?: UseQueryOptions<GroupWithMembers[], ApiError>;
+    queryOptions?: UseQueryOptions<
+        ApiGroupListResponse['results']['data'],
+        ApiError
+    >;
 }) => {
     const setErrorResponse = useQueryError();
-    return useQuery<GroupWithMembers[], ApiError>({
-        queryKey: ['organization_groups', includeMembers],
-        queryFn: () => getOrganizationGroupsQuery(includeMembers),
+    return useQuery<ApiGroupListResponse['results']['data'], ApiError>({
+        queryKey: ['organization_groups', includeMembers, searchInput],
+        queryFn: async () => {
+            return (
+                await getOrganizationGroupsQuery(includeMembers, searchInput)
+            ).data;
+        },
```

Comment:
this is basically an old hook without pagination, right?

Reply:
Yep, this is the exact same request with no pagination, the difference is the returned type since now the endpoint supports pagination we return the data in `results.data`